doc/2/guides/develop-on-kuzzle/2-api-controllers/index.md

```diff
@@ -8,3 +8,452 @@ order: 200
+
+app.controller.use(greetingController)
+```
+
+This way of doing things takes longer to develop but it allows you to have a better code architecture while respecting OOP concepts.
+
+## Handler Function
+
+The handler is the function that will **be called each time our API action is executed**.
+
+This function **takes a [Request object](/core/2/some-link)** as a parameter and **must return a Promise** resolving on the result to be returned to the client.
+
+This function is defined in the `handler` property of an action. Its signature is as follows: `(request: Request) => Promise<any>`.
+
+```js
+app.controller.register('greeting', {
+  actions: {
+    sayHello: {
+      // Handler function for the "greeting:sayHello" action
+      handler: async (request: Request) => {
+        return `Hello, ${request.input.args.name}`
+      }
+    }
+  }
+})
+```
+
+The result returned by our `handler` will be **converted to JSON format** and integrated into the standard Kuzzle response in the `result` property.
+
+```bash
+$ npx wscat -c ws://localhost:7512 --execute '{
+  "controller": "greeting",
+  "action": "sayHello",
+  "name": "Yagmur"
+}'
+
+{
+  "requestId": "a6f4f5b6-1aa2-4cf9-9724-12b12575c047",
+  "status": 200,
+  "error": null,
+  "controller": "greeting",
+  "action": "sayHello",
+  "collection": null,
+  "index": null,
+  "volatile": null,
+  "result": "Hello, Yagmur", # <= handler function return value
+  "room": "a6f4f5b6-1aa2-4cf9-9724-12b12575c047"
+}
+```
+
+## HTTP routes
+
+The execution of an API action through the HTTP protocol is significantly different from other protocols.  
+
+Indeed, **the HTTP protocol uses verbs and routes** in order to address an action whereas the other protocols only use the controller and action name in their JSON payloads.
+
+### Define a HTTP route
+
+When defining a controller action, it is also possible to **specify one or more HTTP routes** available to execute our action using the `http` property.
+
+This property is at the same level as `handler` and **represents an array of routes**.  
+Each route is an object containing a `verb` and a `path` property.
+
+The following HTTP verbs are available: `get`, `post`, `put`, `delete`, `head`.
+
+::: info
+When the `path` property starts with a `/` then the route is added as is, otherwise the route will be prefixed with `/_/`.
+:::
+
+```js
+app.controller.register('greeting', {
+  actions: {
+    sayHello: {
+      handler: async (request: Request) => {
+        return `Hello, ${request.input.args.name}`
+      },
+      http: [
+        // generated route: "GET http://<host>:<port>/greeting/hello"
+        { verb: 'get', path: '/greeting/hello' },
+        // generated route: "POST http://<host>:<port>/_/hello/world"
+        { verb: 'post', path: 'hello/world' },
+      ]
+    }
+  }
+})
+```
+
+::: warning
+It is recommended to let Kuzzle prefix the routes with `/_/` in order not avoid conflict with the existing routes of the standard API.
```

Comment:
```suggestion
It is recommended to let Kuzzle prefix the routes with `/_/` in order to avoid conflict with the existing routes of the standard API.
```